lib/unifex/code_generator/base_types/struct.ex

```diff
@@ -76,7 +76,11 @@ defmodule Unifex.CodeGenerator.BaseTypes.Struct do
     def generate_arg_parse(arg, var_name, ctx) do
       %{postproc_fun: postproc_fun, generator: generator} = ctx
 
-      unique_sufix = String.replace("#{var_name}", ".", "_")
+      unique_sufix =
+        "#{var_name}"
+        # replace array naming
+        |> String.replace("[i]", "")
+        |> String.replace(".", "_")
```

Comment:
@mat-hek agreed this is really ugly (the original replace was as well)
I'd suggest 2 things:
1. modify list parsing to pass a regular var name
```elixir
    @impl true
    def generate_arg_parse(arg, var_name, ctx) do
      elem_name = :"#{var_name}_i"
      len_var_name = "#{var_name}_length"
      native_type = BaseType.generate_native_type(ctx.subtype, ctx.generator, ctx)
      %{subtype: subtype, postproc_fun: postproc_fun, generator: generator} = ctx

      ~g"""
      ({
      int get_list_length_result = enif_get_list_length(env, #{arg}, &#{len_var_name});
      if(get_list_length_result){
        #{var_name} = (#{native_type} *) enif_alloc(sizeof(#{native_type}) * #{len_var_name});

        for(unsigned int i = 0; i < #{len_var_name}; i++) {
          #{native_type} #{elem_name} = #{var_name}[i];
          #{BaseType.generate_initialization(subtype, elem_name, generator, ctx)}
        }

        ERL_NIF_TERM list = #{arg};
        for(unsigned int i = 0; i < #{len_var_name}; i++) {
          ERL_NIF_TERM elem;
          enif_get_list_cell(env, list, &elem, &list);
          #{native_type} #{elem_name} = #{var_name}[i];
          #{BaseType.generate_arg_parse(subtype, elem_name, ~g<elem>, postproc_fun, generator, ctx)}
          #{var_name}[i] = #{elem_name};
        }
      }
      get_list_length_result;
      })
      """
    end
```
  It would be best to do the same in the code, that passes something with a dot as `var_name` to get rid of the need to manipulate `var_name` 
2. create a helper to sanitize variable names replacing special chars (square brackets, dots but also `->`) with an underscore